pkg/utils/eventrecorder/eventrecorder.go

```diff
@@ -65,6 +71,14 @@ func New(rawK8SClient, cachedK8SClient client.Client) (*EventRecorder, error) {
 	eventRecorder.RawK8SClient = rawK8SClient
 	eventRecorder.CachedK8SClient = cachedK8SClient
 
+	if eventRecorder.hostNode, err = findMyNode(eventRecorder.CachedK8SClient); err != nil {
+		log.Errorf("Failed to find host node: %s", err)
+	}
+
+	if eventRecorder.hostPod, err = findMyPod(eventRecorder.CachedK8SClient); err != nil {
```

Comment:
Good call on caching this, this will be helpful for refactoring `BroadcastEvent` as well